### ISSUE TITLE
`no-extra-parens`: Allow when spreading a binary or ternary expression

### ISSUE THREAD

**Issue by lazarljubenovic** [eslint-stylistic/eslint-stylistic#407]
### Clear and concise description of the problem

I want to use `no-extra-parens` rule in general, but allow myself to write 

```ts
const x = [
  ...(a ? [1, 2, 3] : []),
  ...(b ?? c),
]
```

instead of being locked into

```ts
const x = [
  ...a ? [1, 2, 3] : [],
  ...b ?? c,
]
```

so that the precedence is clearer at a glance.

The reasoning is similar to existing `enforceForNewInMemberExpressions` which allows `(new Foo()).bar`, even though it's technically the same as `new Foo().bar`, because `new Foo().bar` looks like `new (Foo().bar)` at a glance to a human reader (mostly due to whitespace).

### Suggested solution

Add a flag like `binaryAndTernaryExpressionsInsideSpread` which will make an exception for this case.

### Alternative

As far as I'm aware, it's impossible to write an additional rule that will “nullify” reports from existing ESLint rules, so I can't just “augment” this with my own rule.

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [x] If this feature request is accepted, I am willing to submit a PR to fix this issue
